File: 22-distributed-orders-kafka/packages/shared/src/env.ts
/** Tiny env reader with defaults pointing at the docker-compose infra. */

type LogLevel = 'trace' | 'debug' | 'info' | 'warn' | 'error' | 'fatal';

export const env = {
  KAFKA_BROKERS: (process.env.KAFKA_BROKERS ?? 'localhost:9092').split(','),
  REDIS_URL: process.env.REDIS_URL ?? 'redis://localhost:6379',
  LOG_LEVEL: (process.env.LOG_LEVEL ?? 'info') as LogLevel,

  GATEWAY_PORT: Number(process.env.GATEWAY_PORT ?? 3000),
  ORDERS_PORT: Number(process.env.ORDERS_PORT ?? 3001),
  PAYMENTS_PORT: Number(process.env.PAYMENTS_PORT ?? 3002),
  INVENTORY_PORT: Number(process.env.INVENTORY_PORT ?? 3003),

  ORDERS_URL: process.env.ORDERS_URL ?? 'http://localhost:3001',

  // The .NET AOT traceability collector — where spans (and later logs) are pushed.
  COLLECTOR_URL: process.env.COLLECTOR_URL ?? 'http://localhost:9317',
};

File: 22-distributed-orders-kafka/services/inventory-py/env.py
"""Tiny env reader with defaults pointing at the docker-compose infra — the Python
mirror of `packages/shared/src/env.ts`. Same broker/redis defaults as the host-side
TypeScript services, so the Python inventory joins the exact same topology."""

import os
from dataclasses import dataclass


@dataclass(frozen=True)
class _Env:
    # aiokafka takes a comma-joined string or list; we keep the raw comma string.
    KAFKA_BROKERS: str = os.getenv("KAFKA_BROKERS", "localhost:9092")
    REDIS_URL: str = os.getenv("REDIS_URL", "redis://localhost:6379")
    LOG_LEVEL: str = os.getenv("LOG_LEVEL", "info")
    INVENTORY_PORT: int = int(os.getenv("INVENTORY_PORT", "3003"))
    # The .NET AOT traceability collector — where spans are pushed.
    COLLECTOR_URL: str = os.getenv("COLLECTOR_URL", "http://localhost:9317")


env = _Env()
